api/tests/src/api/opportunities_v0_1/test_opportunity_route.py

```diff
@@ -987,6 +1093,211 @@ def test_opportunity_search_filters_200(
+                ],
+            ),
+            # Like the previous one, but the query is more specific and only gets one of the scenarios
+            (
+                get_search_request(
+                    page_size=25,
+                    query="very few enum known to humankind",
+                    applicant_type_one_of=[
+                        ApplicantType.STATE_GOVERNMENTS,
+                        ApplicantType.FEDERALLY_RECOGNIZED_NATIVE_AMERICAN_TRIBAL_GOVERNMENTS,
+                    ],
+                ),
+                [Scenario.POSTED_NON_DEFAULT_AGENCY_WITH_APP_TYPES],
+            ),
+            # Agency filtered by one_of, query hits something in the summary description
+            (
+                get_search_request(
+                    page_size=25,
+                    query="i am a description",
+                    agency_one_of=["DIFFERENT-XYZ", "DIFFERENT-abc"],
+                ),
+                [Scenario.CLOSED_NON_DEFAULT_AGENCY_WITH_FUNDING_CATEGORIES],
+            ),
+            ### A few scenarios that don't return any results because filters/query make it too specific
+            (
+                get_search_request(
+                    page_size=25,
+                    query="humankind",
+                    opportunity_status_one_of=[
+                        OpportunityStatus.FORECASTED,
+                        OpportunityStatus.CLOSED,
+                        OpportunityStatus.ARCHIVED,
+                    ],
+                ),
+                [],
+            ),
+            (
+                get_search_request(
+                    page_size=25,
+                    query="different",
+                    funding_instrument_one_of=[FundingInstrument.GRANT],
+                ),
+                [],
+            ),
+            (
+                get_search_request(
+                    page_size=25,
+                    query="words that don't hit anything",
+                    applicant_type_one_of=[
+                        ApplicantType.STATE_GOVERNMENTS,
+                        ApplicantType.FEDERALLY_RECOGNIZED_NATIVE_AMERICAN_TRIBAL_GOVERNMENTS,
+                    ],
+                ),
+                [],
+            ),
+        ],
+        ids=search_scenario_id_fnc,
+    )
+    def test_opportunity_query_and_filter_200(
+        self, client, api_auth_token, search_request, expected_scenarios, setup_scenarios
+    ):
+        # Basically a combo of the above two tests, testing requests with both query text and filters set
+        resp = client.post(
+            "/v0.1/opportunities/search", json=search_request, headers={"X-Auth": api_auth_token}
+        )
+
+        search_response = resp.get_json()
+        assert resp.status_code == 200
+
+        self.validate_results(search_request, search_response, expected_scenarios)
+
+    def validate_results(self, search_request, search_response, expected_scenarios):
```

Comment:
nit: This is a bit of a larger ask but would it make sense to organize things by class - as they can be run that way via console - and try to keep the files under 1000 lines if possible by splitting them up.

Reply:
These are organized by class - what specifically would you suggest be changed?

I do think this set of tests should be tidied up a bit, and splitting the different routes into separate files probably would be the right way to handle that.

Unfortunately, search requires a lot of tests to validate as there are effectively an infinite number of scenarios due to how many different possible values / fields we can search by.

Reply:
> and splitting the different routes into separate files probably would be the right way to handle that.

Yea something like this - I see that some of the parameterized tests use both the singular GET opportunity and plural POST opportunities. It's just early in the project and I would try to avoid the test files ending up like our past project with 5k lines.

Reply:
Made a ticket to do that cleanup: https://github.com/HHS/simpler-grants-gov/issues/1524